### PR TITLE
Add grafana-agent App CR to send PrometheusRules to mimir ruler

### PR DESCRIPTION
**Pull request by marieroque**
Before adding a new alerting rule into this repository you should consider creating an SLO rules instead.
SLO helps you both increase the quality of your monitoring and reduce the alert noise.

* How to create a SLO rule: https://github.com/giantswarm/sloth-rules#how-to-create-a-slo
* Documentation: https://intranet.giantswarm.io/docs/monitoring/slo-alerting/

---
Towards: https://github.com/giantswarm/roadmap/issues/3301

This PR add `grafana-agent` App CR to the chart in order to deploy a grafana-agent in flow mode as a deployment.
Its configuration sends ProemtheusRules to Mimir ruler.

### Checklist

- [x] Update CHANGELOG.md
- [ ] Add [Unit tests](https://github.com/giantswarm/prometheus-rules/#testing)
- [ ] Follow [Alert structure](https://github.com/giantswarm/prometheus-rules/#how-alerts-are-structured)
- [ ] Consider [creating a dashboard](https://docs.giantswarm.io/getting-started/observability/grafana/custom-dashboards/) ([guidelines](https://intranet.giantswarm.io/docs/product/ux/guidelines/dashboards/)) (if it does not exist already) to help oncallers monitor the status of the issue.
- [ ] Request review from oncall area, as well as team (e.g: `oncall-kaas-cloud` GitHub group).

### CODE REVIEW

**Review by QuentinBisson — APPROVED** [review]
LGTM if tested. Could you check if we can build a business alerts as well in case it cannot send rules to Mimir ?

File: helm/prometheus-rules/templates/grafana-agent-rules.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      namespace: ""
+  kubeConfig:
+    context:
+      name: ""
+    inCluster: true
+    secret:
+      name: ""
+      namespace: ""
+  name: grafana-agent
+  namespace: {{ .Values.namespace | quote }}
```
Comment (marieroque) [inline helm/prometheus-rules/templates/grafana-agent-rules.yaml:27]: I couldn't set `mimir` as namespace, I got that error:
```
Reason:         Upgrade "prometheus-rules" failed: failed to create resource: admission webhook "apps.app-admission-controller.giantswarm.io" denied the request: validation error: target namespace mimir is not allowed for in-cluster apps
```

**Comment by QuentinBisson** [comment]
That's a weird limitation again

File: helm/prometheus-rules/templates/grafana-agent-rules.yaml (hunk 1 above)
Comment (QuentinBisson) [reply helm/prometheus-rules/templates/grafana-agent-rules.yaml:27]: That's thé one at the top that Iz causing thé issue L9 should bé giantswarm

Comment (marieroque) [reply helm/prometheus-rules/templates/grafana-agent-rules.yaml:27]: It's the case with `.Release.Namespace` no ?

**Review by QuentinBisson — APPROVED** [review]

**Comment by marieroque** [comment]
It works now :-)
I have to hardcoded `giantswarm` namespace rather than using `.Release.Namespace`

**Review by hervenicol — APPROVED** [review]
LGTM if tested.
Nice to see it implemented in so few lines :star_struck:

File: helm/prometheus-rules/templates/grafana-agent-rules.yaml (hunk 1 above)
Comment (marieroque) [reply helm/prometheus-rules/templates/grafana-agent-rules.yaml]: `monitoring`

**Pull request merged**